File: List1/Ex1/macheps.jl
# Computes machine epsilon. As argument accepts a type (Float16, Float32, Float64)
function macheps(type)
	epsilon = type(1.0)
	one = type(1.0)
	
	while one + epsilon / 2!= one
		epsilon = epsilon / 2
	end
	
	return epsilon
end

File: List1/Ex1/macheps.c
#include <stdio.h>
#include <float.h>

int main(void)
{
    printf("Machine float64 epsilon:     %.64f\n", DBL_EPSILON);
    printf("Machine float32 epsilon:     %.64f\n", FLT_EPSILON);
    return 0;
}
